markdown/DynamicalODE/Henon-Heiles_energy_conservation_benchmark.md

```diff
@@ -0,0 +1,448 @@
+    GC.gc()
+    @time sol4 = solve(prob, Vern9(), abstol=1e-14, reltol=1e-14)
+    GC.gc()
+    @time sol5 = solve(prob, DPRKN12(), abstol=1e-14, reltol=1e-14)
+    GC.gc()
+    (mode == :inplace && all) && @time sol6 = solve(prob, TaylorMethod(50), abstol=1e-20)
+
+    (mode == :inplace && all) && println("Vern9 + ManifoldProjection max energy error:\t"*
+        "$(maximum(abs_energy_err(sol1)))\tin\t$(length(sol1.u))\tsteps.")
+    println("KahanLi8 max energy error:\t\t\t$(maximum(abs_energy_err(sol2)))\tin\t$(length(sol2.u))\tsteps.")
+    println("SofSpa10 max energy error:\t\t\t$(maximum(abs_energy_err(sol3)))\tin\t$(length(sol3.u))\tsteps.")
+    println("Vern9 max energy error:\t\t\t\t$(maximum(abs_energy_err(sol4)))\tin\t$(length(sol4.u))\tsteps.")
+    println("DPRKN12 max energy error:\t\t\t$(maximum(abs_energy_err(sol5)))\tin\t$(length(sol5.u))\tsteps.")
+    (mode == :inplace && all) && println("TaylorMethod max energy error:\t\t\t$(maximum(abs_energy_err(sol6)))"*
+        "\tin\t$(length(sol6.u))\tsteps.")
+
+    if plt === nothing
+        plt = plot(xlabel="t", ylabel="Energy error")
+    end
+    (mode == :inplace && all) && plot!(sol1.t, energy_err(sol1), label="Vern9 + ManifoldProjection")
+    plot!(sol2.t, energy_err(sol2), label="KahanLi8", ls=mode==:inplace ? :solid : :dash)
+    plot!(sol3.t, energy_err(sol3), label="SofSpa10", ls=mode==:inplace ? :solid : :dash)
+    plot!(sol4.t, energy_err(sol4), label="Vern9", ls=mode==:inplace ? :solid : :dash)
+    plot!(sol5.t, energy_err(sol5), label="DPRKN12", ls=mode==:inplace ? :solid : :dash)
+    (mode == :inplace && all) && plot!(sol6.t, energy_err(sol6), label="TaylorMethod")
+
+    return plt
+end
+````
+
+
+````
+compare (generic function with 4 methods)
+````
+
+
+
+
+
+The `mode` argument choses between the in place approach
+and the out of place one. The `all` parameter is used to compare only the integrators that support both
+the in place and the out of place versions (we reffer here only to the 6 high order methods chosen bellow).
+The `plt` argument can be used to overlay the results over a previous plot and the `tmax` keyword determines
+the simulation time.
+
+Note:
+1. The `Vern9` method is used with `ODEProblem` because of performance issues with `ArrayPartition` indexing which manifest for `DynamicalODEProblem`.
+2. The `NLsolve` call used by `ManifoldProjection` was modified to use `ftol=1e-13` in order to obtain a very low energy error.
+
+Here are the results of the comparisons between the in place methods:
+
+````julia
+compare(tmax=1e2)
+````
+
+
+````
+28.477321 seconds (42.07 M allocations: 1.597 GiB, 1.31% gc time)
+  2.993858 seconds (9.36 M allocations: 451.505 MiB, 1.53% gc time)
+  4.171649 seconds (11.93 M allocations: 551.610 MiB, 1.79% gc time)
+  0.196334 seconds (1.07 M allocations: 43.356 MiB)
+  2.035021 seconds (5.77 M allocations: 315.643 MiB, 1.19% gc time)
+Error: type DynamicalODEFunction has no field f
```

Comment:
@SebastianM-C looks like there are some errors.

Reply:
It looks like `TaylorIntegration` doesn't use the automatic conversion from `DynamicalODEProblem` to `ODEProblem`. Should I open an issue for this? I'm not sure where though.

Reply:
Try and fix it. It shouldn't be hard: nothing should be checking and using `f.f` explicitly.